packages/webdriverio/src/commands/browser/downloadFile.ts

```diff
@@ -0,0 +1,87 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import JSZip from 'jszip'
+/**
+ *
+ * Download a file from the remote computer running Selenium node to local file system
+ * by using the [`downloadFile`](https://webdriver.io/docs/api/selenium#downloadFile) command.
+ *
+ * _Note:_ that this command is only supported if you use a
+ * [Selenium Grid](https://www.selenium.dev/documentation/en/grid/) with Chrome, Edge or Firefox.
+ *
+ * <example>
+ :downloadFile.js
+ it('should download a file', async () => {
+        // capabilities: [
+        //  {
+        //      browserName: 'chrome',
+        //      se:downloadsEnabled': true
+        //  }]
+
+        await browser.url('https://www.selenium.dev/selenium/web/downloads/download.html')
+
+        await $('#file-1').click()
+
+        await browser.waitUntil(async function () {
+            return (await browser.getDownloadableFiles()).names.includes('file_1.txt')
+        }, {timeout: 5000})
+
+        const files = await browser.getDownloadableFiles()
+
+        await browser.downloadFile(files.names[0], process.cwd())
+
+        await browser.deleteDownloadableFiles()
+    })
+ * </example>
+ *
+ * @alias browser.downloadFile
+ * @param {string} fileName remote path to file
+ * @param {string} targetDirectory target location on local computer
+ * @type utility
+ * @uses protocol/download
+ *
+ */
+export async function downloadFile(
+    this: WebdriverIO.Browser,
+    fileName: string,
+    targetDirectory: string
+) {
+    /**
+     * parameter check
+     */
+    if (typeof fileName !== 'string' || typeof targetDirectory !== 'string') {
+        throw new Error('number or type of arguments don\'t agree with downloadFile command')
+    }
+
+    /**
+     * check if command is available
+     */
+    if (typeof this.download !== 'function') {
+        throw new Error(`The downloadFile command is not available in ${(this.capabilities as WebdriverIO.Capabilities).browserName}`)
+    }
+
+    const response = await this.download(fileName)
+    const base64Content = response.contents
+
+    if (!targetDirectory.endsWith('/')) {
+        targetDirectory += '/'
+    }
+
+    fs.mkdirSync(targetDirectory, { recursive: true })
+    const zipFilePath = path.join(targetDirectory, `${fileName}.zip`)
+    fs.writeFileSync(zipFilePath, Buffer.from(base64Content, 'base64'))
+
+    const zipData = fs.readFileSync(zipFilePath)
+    await JSZip.loadAsync(zipData)
+        .then((zip) => {
+            // Iterate through each file in the zip archive
+            Object.keys(zip.files).forEach(async (fileName) => {
+                const fileData = await zip.files[fileName].async('nodebuffer')
+                fs.writeFileSync(`${targetDirectory}/${fileName}`, fileData)
+                console.log(`File extracted: ${fileName}`)
+            })
+        })
+        .catch((error) => {
+            console.error('Error unzipping file:', error)
+        })
+}
```

Comment:
Can we return a map with files paths so the user has an easier time to e.g. load the file contents in the test?